### PR TITLE
Add guidance for sbom generation through arcade templates

### PR DESCRIPTION
**Pull request by riarenas**
https://github.com/dotnet/core-eng/issues/15480

### To double check:

* [ ] The right tests are in and and the right validation has happened.  Guidance:  https://github.com/dotnet/core-eng/tree/master/Documentation/Validation

### CODE REVIEW

File: Documentation/SBOMGenerationGuidance.md [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+to encapsulate the creation and upload of the SBOM to the build's artifacts,
+leveraging the [ADO SBOM generator
+task](https://eng.ms/docs/cloud-ai-platform/devdiv/one-engineering-system-1es/1es-docs/secure-supply-chain/ado-sbom-generator).
+This section describes how to integrate this template into your repo's official
+build depending on your current usage of the YAML templates that Arcade
+provides.
+
+### Repositories using Arcade's job(s).yml templates
+
+If you are using the Arcade Job templates
```
Comment (mmitche) [inline Documentation/SBOMGenerationGuidance.md:27]: I would specify arcade channels/versions that will be supporting this functionality here (6.0 and main)

File: Documentation/SBOMGenerationGuidance.md [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+  Arcade's convention. 
+- `sbomContinueOnError`: By default the tasks are set up to not break the build
+  and instead continue on error.
+
+
+## Reviewing generated SBOMs for correctness
+
+To verify that the functionality worked as expected in your repository:
+
+1. The following steps should be added to every job that is using the templates:
```
Comment (mmitche) [inline Documentation/SBOMGenerationGuidance.md:95]:
```suggestion
1. The following steps should have been added to every job that is using the templates:
```

File: Documentation/SBOMGenerationGuidance.md [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+
+- `Build outputs`: Thsese should be present in the `files` collection of the
+  manifest. At this time, we are not concerned with the tooling overreporting
+  the build outputs, but the SBOMs should be checked for missing output
+  packages. 
+
+- `Dependencies`: These should be present in the `packages` collection of the
+  manifest, and use the same detection mechanism as the component governance
+  tasks. 
+
```
Comment (mmitche) [inline Documentation/SBOMGenerationGuidance.md:123]: I'd add a bit here about verifying the product name and version.

**Review by mmitche — APPROVED** [review]
A few minor issues but overall looks g2g

**Comment by riarenas** [comment]
I know this is automated but:

> [riarenas](https://github.com/riarenas) dismissed [mmitche](https://github.com/mmitche)’s [stale review](https://github.com/dotnet/arcade/pull/8450#pullrequestreview-878079241) via [960ff59](https://github.com/dotnet/arcade/commit/960ff590d6edcb55a4fe148104e99710b2b5289f) [1 minute ago](https://github.com/dotnet/arcade/pull/8450#event-6037529008)

I did no such thing.

**Comment by mmitche** [comment]
> I know this is automated but:
> 
> > [riarenas](https://github.com/riarenas) dismissed [mmitche](https://github.com/mmitche)’s [stale review](https://github.com/dotnet/arcade/pull/8450#pullrequestreview-878079241) via [960ff59](https://github.com/dotnet/arcade/commit/960ff590d6edcb55a4fe148104e99710b2b5289f) [1 minute ago](https://github.com/dotnet/arcade/pull/8450#event-6037529008)
> 
> I did no such thing.

It's the new push that does it.

**Comment by riarenas** [comment]
Yep, I just don't appreciate the wording 😄

**Comment by markwilkie** [comment]
Yea, the wording is weird.  It should have said that "riarenas was dismissive of mmitche's old and irrelevant review"

**Comment by riarenas** [comment]
That does sound more like me. If GitHub is going to impersonate me it should do a better job.

File: Documentation/SBOMGenerationGuidance.md (hunk 1 above)
Comment (riarenas) [reply Documentation/SBOMGenerationGuidance.md]: Added a blurb. Let me know if you think I should go more in depth.

File: Documentation/SBOMGenerationGuidance.md (hunk 3 above)
Comment (riarenas) [reply Documentation/SBOMGenerationGuidance.md:139]: Added.

**Review by mmitche — APPROVED** [review]

**Pull request merged**